packages/components/src/components/AutoScalingInput/AutoScalingInput.tsx

```diff
@@ -60,32 +60,34 @@ const StyledInput = styled.input`
  */
 export const AutoScalingInput = forwardRef<HTMLInputElement, Props>(
     ({ value, minWidth, ...props }, ref) => {
-        const [placeholderWidth, setPlaceholderWidth] = useState(0);
+        const [calculatedMin, setCalculatedMin] = useState(minWidth);
         const inputRef = useRef<HTMLInputElement | null>(null);
         const placeholderMeasureRef = useRef<HTMLInputElement | null>(null);
 
-        const calculatedMin = Math.max(minWidth, placeholderWidth);
+       console.log('yay', inputRef?.current?.scrollWidth) 
+
 
-        // Measure the size of the placeholder
+        // Measure the size of the placeholder 
         useEffect(() => {
             if (placeholderMeasureRef?.current?.style) {
-                setPlaceholderWidth(placeholderMeasureRef.current.scrollWidth ?? 0);
+                setCalculatedMin(Math.max(minWidth, placeholderMeasureRef.current.scrollWidth ?? 0));
+                console.log('fdfd', placeholderMeasureRef.current.scrollWidth)
             }
-        }, [setPlaceholderWidth]);
+        });
```

Comment:
missing dependecy: `, [setCalculatedMin]`

Reply:
As I wrote in the message in Slack, this is a complete mess and all of it will have to change. It works somehow currently but I'd like to understand why.

Reply:
yep, I am just making notes so we dont forget anything